### PR TITLE
Move context value provider from MediaProvider to local/useContextValueProvider

### LINKED ISSUE #2425: Move 'local' context value from MediaProvider.js to local/useContextStateProvider.js
## Task Description

This a simple move of code to follow the folder structure described in https://github.com/google/web-stories-wp/issues/2423.

### PR DESCRIPTION
**Pull request by cvolzke4**
## Summary

Move the body of MediaProvider that generates the context value to media/local/provideContextValue.
As per: https://docs.google.com/document/d/1S9jTh6Lz3UEhj9Wkb7S7x_kJGvRVpEaNOqk2aCyb4y8.

## Testing Instructions

Open the media pane to verify that it appears.

### CODE REVIEW

File: assets/src/edit-story/app/media/local/provideContextValue.js [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+/**
+ * Internal dependencies
+ */
+import { useAPI, useConfig } from '../..';
+import useUploadVideoFrame from '../utils/useUploadVideoFrame';
+import useUploadMedia from '../useUploadMedia';
+import { getResourceFromAttachment } from '../utils';
+
+/* eslint-disable react-hooks/rules-of-hooks */
```
Comment (cvolzke4) [inline assets/src/edit-story/app/media/local/provideContextValue.js]: provideContextValue is clearer than the alternatives, eg:
- useContextValue() - may be confused with useContext(), which is a consumer.
- useContextValueProvider() - doesn't return a provider fn, the value itself is returned.
- useProvideContextValue() - doesn't form a valid sentence.

**Review by joannag6 — APPROVED** [review]
LGTM on provideContextValue name change

File: assets/src/edit-story/app/media/local/provideContextValue.js (hunk 1 above)
Comment (cvolzke4) [reply assets/src/edit-story/app/media/local/provideContextValue.js]: This has been renamed to be useContextValueProvider.

**Comment by cvolzke4** [comment]
> LGTM on provideContextValue name change

👍

**Pull request merged**